src/explore-education-statistics-common/src/modules/table-tool/components/__tests__/TableToolWizard.test.tsx

```diff
@@ -311,4 +320,148 @@ describe('TableToolWizard', () => {
+    tableBuilderService.getSubjectMeta.mockResolvedValue(testSubjectMeta);
+    tableBuilderService.filterSubjectMeta.mockResolvedValue(testSubjectMeta);
+    render(
+      <TableToolWizard
+        themeMeta={testThemeMeta}
+        initialState={{
+          initialStep: 2,
+          subjects: testSubjects,
+          query: {
+            publicationId: 'publication-1',
+            subjectId: 'subject-1',
+            locations: {
+              localAuthority: ['barnet'],
+            },
+            filters: ['ethnicity-major-asian-total'],
+            indicators: ['authorised-absence-sessions'],
+            timePeriod: {
+              endCode: 'AY',
+              endYear: 2013,
+              startCode: 'AY',
+              startYear: 2014,
+            },
+          },
+        }}
+      />,
+    );
+
+    // Change subject at Step 2
+    const subjectRadios = screen.getAllByLabelText(/Subject/);
+    userEvent.click(subjectRadios[1]);
+
+    userEvent.click(screen.getByRole('button', { name: 'Next step' }));
+
+    // Step 3
+    await waitFor(() => {
+      expect(screen.getByText('Step 3')).toBeInTheDocument();
+    });
+
+    const localAuthorityCheckboxes = within(
+      screen.getByRole('group', {
+        name: 'Local authority',
+        hidden: true,
+      }),
+    ).getAllByRole('checkbox', {
+      hidden: true,
+    });
+    expect(localAuthorityCheckboxes[0]).not.toBeChecked();
+    expect(localAuthorityCheckboxes[1]).not.toBeChecked();
+
+    userEvent.click(localAuthorityCheckboxes[0]);
+    userEvent.click(screen.getByRole('button', { name: 'Next step' }));
+
+    // Step 4
+    await waitFor(() => {
+      expect(screen.getByText('Step 4')).toBeInTheDocument();
+    });
+
+    const startDateOptions = within(
+      screen.getByLabelText('Start date'),
+    ).getAllByRole('option') as HTMLOptionElement[];
+    expect(startDateOptions[0].selected).toBe(true);
+
+    const endDateOptions = within(
+      screen.getByLabelText('End date'),
+    ).getAllByRole('option') as HTMLOptionElement[];
+    expect(endDateOptions[0].selected).toBe(true);
+
+    userEvent.selectOptions(screen.getByLabelText('Start date'), ['2013_AY']);
+    userEvent.selectOptions(screen.getByLabelText('End date'), ['2013_AY']);
+    userEvent.click(screen.getByRole('button', { name: 'Next step' }));
+
+    // Step 5
+    await waitFor(() => {
+      expect(screen.getByText('Step 5')).toBeInTheDocument();
+    });
+
+    const filterCheckboxes = within(
+      screen.getByRole('group', {
+        name: 'Characteristic',
+        hidden: true,
+      }),
+    ).getAllByRole('checkbox', {
+      hidden: true,
+    });
+    expect(filterCheckboxes[0]).not.toBeChecked();
+
+    const indicatorCheckboxes = within(
+      screen.getByRole('group', {
+        name: 'Indicators',
+      }),
+    ).getAllByRole('checkbox', {});
+    expect(indicatorCheckboxes[0]).not.toBeChecked();
```

Comment:
This is interesting. Seems like if we're clearing time periods, we should also put in some stuff to clear away any filters and indicators too when submitting the time period step?

Reply:
Not sure what you mean, this test relates to resetting all the subsequent steps when you change the subject. It's only when the subject changes that this should happen, if you change the time period the filters and indicators should still be selected.

Reply:
I mean that we've put in some logic to only select time periods if they still exist in subject meta once it's been changed (when moving from locations -> time periods), however, it doesn't look like we've put in anything to remove any filter/indicator options when moving from time periods -> filters/indicators.

On the basis that we're filtering out time periods, it might make sense to also do this for filters/indicators too?